src/app/qgssnappingwidget.cpp

```diff
@@ -113,6 +113,25 @@ QgsSnappingWidget::QgsSnappingWidget( QgsProject *project, QgsMapCanvas *canvas,
   mEnabledAction->setObjectName( QStringLiteral( "EnableSnappingAction" ) );
   connect( mEnabledAction, &QAction::toggled, this, &QgsSnappingWidget::enableSnapping );
 
+  // avoid intersection mode button
+  mAvoidIntersectionsModeButton = new QToolButton();
+  mAvoidIntersectionsModeButton->setToolTip( tr( "Avoid Overlap Mode" ) );
```

Comment:
We could even go a bit further here I think.

```suggestion
  mAvoidIntersectionsModeButton->setToolTip( tr( "When avoid overlap is enabled, digitized features will be clipped to not overlapped existing ones." ) );
```